src/runtime-rs/crates/shim/src/shim_delete.rs

```diff
@@ -41,9 +44,28 @@ impl ShimExecutor {
             info!(sl!(), "remote socket path: {:?}", &file_path);
             fs::remove_file(file_path).ok();
         }
-        service::ServiceManager::cleanup(&self.args.id)
-            .await
-            .context("cleanup")?;
+
+        if let Err(e) = service::ServiceManager::cleanup(&self.args.id).await {
+            error!(
+                sl!(),
+                "failed to cleanup in service manager: {:?}. force shutdown shim process", e
+            );
+
+            let bundle_path = get_bundle_path().context("get bundle path")?;
+            if let Ok(spec) = load_oci_spec() {
+                if let Ok(ContainerType::PodSandbox) = get_contaier_type(&spec) {
+                    // only force shutdown for sandbox container
+                    if let Ok(shim_pid) = self.read_pid_file(&bundle_path) {
```

Comment:
Is it necessary to invalidate `self` after the kill (something like `self.shutdown = true`? If so, that would make it easier to unit test (particularly if the method was specified as:

```rust
fn force_shutdown(&self, signal: nix::signal::Signal) { ... }